### PR TITLE
add method == true in base.py

### PR DESCRIPTION
**Pull request by Lenoush**
add 

https://github.com/mind-inria/mri-nufft/blob/504426774146b209ea7fb2f67133022a04748dbb/src/mrinufft/operators/base.py#L344-L349

### CODE REVIEW

File: src/mrinufft/operators/base.py [hunk 1]
```diff
@@ -332,20 +332,24 @@ def compute_density(self, method=None):
...
-            return None
         if not method:
             self.density = None
             return None
+        if method is True and hasattr(self, "pipe"):
+            method = "pipe"
+
+        if isinstance(method, np.ndarray):
+            self.density = method
+            return None
```
Comment (paquiteau) [inline src/mrinufft/operators/base.py:352]: no need to move this

File: src/mrinufft/operators/base.py [hunk 2]
```diff
@@ -332,20 +332,24 @@ def compute_density(self, method=None):
...
+            If a bool, it will enable or disable the density compensation.
         """
-        if isinstance(method, np.ndarray):
-            self.density = method
-            return None
         if not method:
             self.density = None
             return None
+        if method is True and hasattr(self, "pipe"):
+            method = "pipe"
```
Comment (paquiteau) [inline src/mrinufft/operators/base.py:348]: you can remove the hasattr stuff, `method="pipe"` will use the `isinstance(method,str)` path, and will raise an error done the line if its not available for the backend.

File: src/mrinufft/operators/base.py [hunk 3]
```diff
@@ -332,20 +332,23 @@ def compute_density(self, method=None):
...
             If an array, it should be of shape (Nsamples,) and will be used as is.
+            If a bool, it will enable or disable the density compensation.
         """
         if isinstance(method, np.ndarray):
             self.density = method
             return None
         if not method:
             self.density = None
             return None
+        if method is True:
```
Comment (chaithyagr) [inline src/mrinufft/operators/base.py:350]:
```suggestion
        if method is True and self.backend is in ['tfnufft', 'gpunufft', 'torchkbnufft']:
```

As we dont support `pipe` for other classes.

File: src/mrinufft/operators/base.py [hunk 4]
```diff
@@ -332,20 +332,23 @@ def compute_density(self, method=None):
...
         ----------
-        method: str or callable or array or dict
+        method: str or callable or array or dict or bool
             The method to use to compute the density compensation.
             If a string, the method should be registered in the density registry.
             If a callable, it should take the samples and the shape as input.
             If a dict, it should have a key 'name', to determine which method to use.
             other items will be used as kwargs.
             If an array, it should be of shape (Nsamples,) and will be used as is.
+            If a bool, it will enable or disable the density compensation.
```
Comment (chaithyagr) [inline src/mrinufft/operators/base.py:342]:
```suggestion
            If `True`, the method `pipe` is chosen as default estimation method, if 
            `backend` is `tfnufft`, `gpunufft` or `torchkbnufft`
```

**Review by chaithyagr — CHANGES_REQUESTED** [review]
Minor issues

File: src/mrinufft/operators/base.py [hunk 5]
```diff
@@ -334,20 +334,28 @@ def compute_density(self, method=None):
...
             return None
         if not method:
             self.density = None
             return None
+        if method is True and self.backend in [
+            "tensorflow",
+            "gpunufft",
+            "torchkbnufft",
+        ]:
+            method = "pipe"
```
Comment (paquiteau) [inline src/mrinufft/operators/base.py:358]:
```suggestion
        if method is True:
            method = "pipe"
```

**Review by paquiteau — APPROVED** [review]

**Pull request merged**